speech/grpc/src/main/java/com/google/cloud/speech/grpc/demos/AsyncRecognizeClient.java

```diff
@@ -0,0 +1,248 @@
+//
+// Then set environment variable GOOGLE_APPLICATION_CREDENTIALS to the full path of that file.
+
+package com.google.cloud.speech.grpc.demos;
+
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.speech.v1beta1.AsyncRecognizeRequest;
+import com.google.cloud.speech.v1beta1.AsyncRecognizeResponse;
+import com.google.cloud.speech.v1beta1.RecognitionAudio;
+import com.google.cloud.speech.v1beta1.RecognitionConfig;
+import com.google.cloud.speech.v1beta1.RecognitionConfig.AudioEncoding;
+import com.google.cloud.speech.v1beta1.SpeechGrpc;
+
+import com.google.longrunning.GetOperationRequest;
+import com.google.longrunning.Operation;
+import com.google.longrunning.OperationsGrpc;
+
+import io.grpc.ManagedChannel;
+import io.grpc.StatusRuntimeException;
+import io.grpc.auth.ClientAuthInterceptor;
+import io.grpc.netty.NegotiationType;
+import io.grpc.netty.NettyChannelBuilder;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.OptionBuilder;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Client that sends audio to Speech.AsyncRecognize and returns transcript.
+ */
+public class AsyncRecognizeClient {
+
+  private static final Logger logger =
+      Logger.getLogger(AsyncRecognizeClient.class.getName());
+
+  private static final List<String> OAUTH2_SCOPES =
+      Arrays.asList("https://www.googleapis.com/auth/cloud-platform");
+
+  private final String host;
+  private final int port;
+  private final URI input;
+  private final int samplingRate;
+
+  private final ManagedChannel channel;
+  private final SpeechGrpc.SpeechBlockingStub stub;
+  private final OperationsGrpc.OperationsBlockingStub statusStub;
+
+  /**
+   * Construct client connecting to Cloud Speech server at {@code host:port}.
+   */
+  public AsyncRecognizeClient(String host, int port, URI input, int samplingRate)
+      throws IOException {
+    this.host = host;
+    this.port = port;
+    this.input = input;
+    this.samplingRate = samplingRate;
+
+    GoogleCredentials creds = GoogleCredentials.getApplicationDefault();
+    creds = creds.createScoped(OAUTH2_SCOPES);
+    channel = NettyChannelBuilder.forAddress(host, port)
+        .negotiationType(NegotiationType.TLS)
+        .intercept(new ClientAuthInterceptor(creds, Executors.newSingleThreadExecutor()))
+        .build();
+    stub = SpeechGrpc.newBlockingStub(channel);
+    statusStub = OperationsGrpc.newBlockingStub(channel);
+
+    logger.info("Created stub for " + host + ":" + port);
+  }
+
+  private RecognitionAudio createRecognitionAudio() throws IOException {
```

Comment:
This function seems unnecessary. I'd rather see the RecognitionAudioFactory call whereever this is used.